marshal/claim.go

```diff
@@ -311,7 +314,16 @@ func (c *claim) messagePump() {
 		c.lock.Unlock()
 
 		// Push the message down to the client (this bypasses the Consumer)
-		c.messages <- msg
+		// We should NOT write to the consumer channel if the claim is no longer claimed. This
+		// needs to be serialized with Release, otherwise a race-condition can potentially
+		// lead to a write to a closed-channel. That's why we're using this lock. We're not
+		// using the main lock to avoid deadlocks since the write to the channel is blocking
+		// until someone consumes the message blocking all Commit operations
+		c.messagesLock.Lock()
```

Comment:
I thought about this and couldn't think of a better way of solving this...the main problem is that the creator of the channel (the consumer) is not the one writing to it....this is the current design....
I'll be happy to change this if you managed to think of something over the weekend